apis/rust/node/src/daemon/mod.rs

```diff
@@ -274,37 +274,43 @@ impl EventStream {
                         continue;
                     }
                 };
-                let drop_token = match &event {
-                    NodeEvent::Input {
-                        data: Some(data), ..
-                    } => data.drop_token(),
-                    NodeEvent::Stop
-                    | NodeEvent::InputClosed { .. }
-                    | NodeEvent::Input { data: None, .. } => None,
-                };
+                for event in events {
+                    let drop_token = match &event {
+                        NodeEvent::Input {
+                            data: Some(data), ..
+                        } => data.drop_token(),
+                        NodeEvent::Stop
+                        | NodeEvent::InputClosed { .. }
+                        | NodeEvent::Input { data: None, .. } => None,
+                    };
 
-                let (drop_tx, drop_rx) = std::sync::mpsc::channel();
-                match tx.send(EventItem::NodeEvent {
-                    event,
-                    ack_channel: drop_tx,
-                }) {
-                    Ok(()) => {}
-                    Err(_) => {
-                        // receiving end of channel was closed
-                        break Ok(());
+                    let (drop_tx, drop_rx) = std::sync::mpsc::channel();
+                    match tx.send(EventItem::NodeEvent {
+                        event,
+                        ack_channel: drop_tx,
+                    }) {
+                        Ok(()) => {}
+                        Err(_) => {
+                            // receiving end of channel was closed
+                            break 'outer Ok(());
+                        }
                     }
-                }
 
-                match drop_rx.recv_timeout(Duration::from_secs(30)) {
-                    Ok(()) => break Err(eyre!("Node API should not send anything on ACK channel")),
-                    Err(std::sync::mpsc::RecvTimeoutError::Timeout) => {
-                        tracing::warn!("timeout while waiting for input ACK");
+                    match drop_rx.recv_timeout(Duration::from_secs(30)) {
+                        Ok(()) => {
+                            break 'outer Err(eyre!(
+                                "Node API should not send anything on ACK channel"
+                            ))
+                        }
+                        Err(std::sync::mpsc::RecvTimeoutError::Timeout) => {
+                            tracing::warn!("timeout while waiting for input ACK");
```

Comment:
```suggestion
                            tracing::warn!("timeout after waiting 30 seconds for input ACK");
```
or:
```suggestion
                            tracing::warn!("timeout after waiting {:#?} for input ACK", ack_timeout_duration);
```
with `ack_timeout_duration` the hard-coded value of this timeout could help debug in case something goes wrong.